### PR TITLE
chore: removed deprecated kubelet executor. Fixes #7803

### LINKED ISSUE #7803: Remove `kubelet` executor
Not needed, use Emissary.

### PR DESCRIPTION
**Pull request by rohankmr414**
Signed-off-by: Rohan Kumar <rohankmr414@gmail.com>

Fixes #7803

### CODE REVIEW

**Comment by alexec** [comment]
You've some linting errors.

**Comment by blkperl** [comment]
Hi @rohankmr414,

Thanks for the PR, I found a few more things to remove.

- `.github/workflows/ci-build.yaml` needs to have kubelet removed
- `workflow/validate/validate.go` has a reference to the kubelet executor in a comment for `ContainerRuntimeExecutor`
- There are some quickstarts for kubelet in manifests/quick-start/base/executor/kubelet 
- `KubeletPort` and KubeletInsecure is a config only needed by the kubelet executor and can be removed from the config/code/docs (config/config.go & workflow/common/common.go and other places)

File: test/e2e/fixtures/needs.go [hunk 1]
```diff
@@ -10,13 +10,8 @@ var (
 	CI Need = func(s *E2ESuite) (bool, string) {
 		return os.Getenv("CI") != "", "CI"
 	}
-	BaseLayerArtifacts Need = func(s *E2ESuite) (bool, string) {
-		met, _ := None(Kubelet)(s)
-		return met, "base layer artifact support"
-	}
 	Docker   = Executor("docker")
 	Emissary = Executor("emissary")
-	Kubelet  = Executor("kubelet")
```
Comment (blkperl) [inline test/e2e/fixtures/needs.go:19]: There are a few more

```
test/e2e/argo_server_test.go:           s.Need(fixtures.None(fixtures.Kubelet))
test/e2e/artifacts_test.go:     s.Need(fixtures.None(fixtures.Docker, fixtures.Kubelet))
```

**Comment by rohankmr414** [comment]
@blkperl Thank you for pointing out the errors. I have made the necessary changes. Can you please review it?

**Review by alexec — DISMISSED** [review]
Can we leave Kubernetes executor role example? It’s docs.

**Review by terrytangyuan — APPROVED** [review]
LGTM

File: config/config.go [hunk 2]
```diff
@@ -39,12 +39,6 @@ type Config struct {
 
 	ContainerRuntimeExecutors ContainerRuntimeExecutors `json:"containerRuntimeExecutors,omitempty"`
 
-	// KubeletPort is needed when using the kubelet containerRuntimeExecutor, default to 10250
-	KubeletPort int `json:"kubeletPort,omitempty"`
```
Comment (terrytangyuan) [inline config/config.go:43]: Could you also update the example configmap in docs folder to add a comment saying that these will be deprecated?

Comment (rohankmr414) [reply config/config.go:43]: done

**Pull request merged**